app/services/host.py

```diff
@@ -90,9 +101,10 @@ def update(self, service, ip_address, service_repo_name, port, revision, last_ch
         return True
 
     def set_tag(self, service, ip_address, tag_name, tag_value):
-        host = Host.get(service, ip_address)
-        host.tags[tag_name] = tag_value
-        host.save()
+        # TODO note that we never sweep when we do a get... is that an error?
```

Comment:
Hm, so I guess the issue is that it is unclear what should happen if the host is expired... the thing that makes the most sense to me is to return True if the update happened, False otherwise, but that'd change the semantics of the API a bit. If we do that, I think we should just be consistent in doing it that way (I'm happy to add to the docs, etc). Does that sound reasonable to you?